e2e/nomostest/clusters/gke.go

```diff
@@ -260,6 +265,33 @@ func createGKECluster(t testing.NTB, name string) error {
 	return listAndWaitForOperations(context.Background(), t, name)
```

Comment:
The `listAndWaitForOperations` call at the end of `createGKECluster` and `getGKECredentials` is currently intended to serve as a readiness check. I'd suggest either:

1. Instead of surfacing a new `WaitForReady` interface method, replace the `listAndWaitForOperations` calls with `waitForReadyGKECluster`, assuming `listAndWaitForOperations` is insufficient to determine readiness.
2. Keep the new `WaitForReady` interface method but remove the existing readiness checks built in to `createGKECluster`/`getGKECredentials`